### PR TITLE
[prototype only] ports react

### PR DESCRIPTION
**Pull request by jlsnow301**
## About The Pull Request
Long term project I've been working on to port TGUI into react / vite. You can see the top-end approach here, which still has problems. It does not seem to be able to get the backend state.
## Why It's Good For The Game
If we were on React I could shave down a ton of tgui code and probably make the system faster as a result
## Changelog
N/A

### CODE REVIEW

File: tgui/packages/tgui/components/Popper.tsx [hunk 1]
```diff
@@ -43,7 +46,7 @@ export class Popper extends Component<PopperProps> {
       // This code is copied from `findDOMNode` in inferno-extras.
       // Because this component is written in TypeScript, we will know
       // immediately if this internal variable is removed.
```
Comment (Mothblocks) [inline tgui/packages/tgui/components/Popper.tsx:48]: youll want to remove this entire hack comment

Comment (Mothblocks) [reply tgui/packages/tgui/components/Popper.tsx:48]: adn then also make sure this actually works lol idk if this was a fundamental issue with refs or if it was just an infenro bug

**Comment by Mothblocks** [comment]
react *ought* to give us performance gains but the primary reason is because hooks are goated

File: tgui/packages/tgui/components/Box.tsx [hunk 2]
```diff
@@ -264,15 +263,5 @@ export const Box: SFC<BoxProps> = (props: BoxProps) => {
...
-  return createVNode(
-    VNodeFlags.HtmlElement,
-    as,
-    computedClassName,
-    children,
-    ChildFlags.UnknownChildren,
-    computedProps,
-    undefined
-  );
+  return createElement(as, computedClassName, children, undefined);
```
Comment (jlsnow301) [inline tgui/packages/tgui/components/Box.tsx:266]: i'm suspicious of this block in particular

Comment (Mothblocks) [reply tgui/packages/tgui/components/Box.tsx:266]: so this would instead be `createElement(as, { className: computedClassName }, children)`

**Comment by optimumtact** [comment]
webshit

**Comment by jlsnow301** [comment]
this will need the state manager changed (which I am willing to do, but will take time) so I will close this for now

**Pull request closed without merging**